views/reports/half_year.php

```diff
@@ -1,114 +1,231 @@
+	</div>
+	<div class="container py-3">
+		<table class="table table-active">
+			<thead>
+			</thead>
+			<tbody>
+				<tr>
+					<th scope="row">Total # of clients served (unduplicated):</td>
+						<td><b><?=$pResult?></b></td>
+				</tr>
+			</tbody>
+		</table>
+	</div>
+	<div class="container pb-2">
+		<h3 class="text-center">Survey Results</h3>
+	</div>
+	<div class="container">
+		<table class="table table-hover">
+			<thead>
+				<tr>
+					<th>Questions</th>
+					<th>% in Favor</th>
+					<th># in Favor</th>
+					<th>Total Respondents</th>
+				</tr>
+			</thead>
+			<tbody>
+				<tr>
+					<td scope="row">Have an increased knowledge of the topics</td>
+					<td>
+						<?php if (count($sFavorResults) > 0) {
+						if ($sTotalResults["topic"] > 0) 
+							echo ($sFavorResults["topic"]/$sTotalResults["topic"])*100 . "%";
+					}?>
+					</td>
+					<td>
+						<?php if (count($sFavorResults) > 0) echo $sFavorResults["topic"];?>
+					</td>
+					<td>
+						<?php if (count($sTotalResults) > 0) echo $sTotalResults["topic"];?>
+					</td>
+				</tr>
+				<tr>
+					<td scope="row">Plan on using specific techniques discussed in class</td>
+					<td>
+						<?php if (count($sFavorResults) > 0) {
+						if ($sTotalResults["practice"] > 0) 
+							echo ($sFavorResults["practice"]/$sTotalResults["practice"])*100 . "%";
+					}?>
+					</td>
+					<td>
+						<?php if (count($sFavorResults) > 0) echo $sFavorResults["practice"];?>
+					</td>
+					<td>
+						<?php if (count($sTotalResults) > 0) echo $sTotalResults["practice"];?>
+					</td>
+				</tr>
+				<tr>
+					<td scope="row">Realized other parents share the same concerns</td>
+					<td>
+						<?php if (count($sFavorResults) > 0) {
+						if ($sTotalResults["otherparents"] > 0) 
+							echo ($sFavorResults["otherparents"]/$sTotalResults["otherparents"])*100 . "%";
+					}?>
+					</td>
+					<td>
+						<?php if (count($sFavorResults) > 0) echo $sFavorResults["otherparents"];?>
+					</td>
+					<td>
+						<?php if (count($sTotalResults) > 0) echo $sTotalResults["otherparents"];?>
+					</td>
+				</tr>
+				<tr>
+					<td scope="row">Understand children have different perspectives than they do</td>
+					<td>
+						<?php if (count($sFavorResults) > 0) {
+						if ($sTotalResults["perspective"] > 0) 
+							echo ($sFavorResults["perspective"]/$sTotalResults["perspective"])*100 . "%";
+					}?>
+					</td>
+					<td>
+						<?php if (count($sFavorResults) > 0) echo $sFavorResults["perspective"];?>
+					</td>
+					<td>
+						<?php if (count($sTotalResults) > 0) echo $sTotalResults["perspective"];?>
+					</td>
+				</tr>
+			</tbody>
+		</table>
+	</div>
 </div>
+<script>
+	const NUM_YEARS_BACK = 4;
+
+	window.onload = initPage;
+
+	function initPage() {
+		var d = new Date();
+		var halfElem = document.getElementById("half");
+		var yearElem = document.getElementById("year");
+		var year = d.getFullYear();
+		for (i = 0; i <= NUM_YEARS_BACK; i++) {
```

Comment:
@algozine had liked the idea of a rolling 4 year window. Not sure it's the best solution since you'd be missing out on certain reports eventually, on the other hand the dropdown option list could get very long.

Reply:
Yes, I'm leaving this as is for now because Algozzine specifically asked for it. Though it might be a question for the Gwen..